classes/form/cms_types_form.php

```diff
@@ -100,30 +105,6 @@ protected function definition() {
         $this->add_action_buttons();
     }
 
-    /**
-     * Extra validataion on the data.
-     *
-     * @param \stdClass $data
-     * @param array $files
-     * @param array $errors
-     * @return array
-     */
-    public function extra_validation($data, $files, array &$errors) {
```

Comment:
Why are we removing validation here?

Also shouldn't' we add validation that the idnumber is not currently used?

Reply:
I added the relevant functions to cms_types. I believe they are better there.

Reply:
https://github.com/catalyst/moodle-mod_cms/pull/92/files#diff-53c6c2616dd05e8d131ed7cc22c92e2d18fccbc96a79f7923e2c6604c78672a7R144
https://github.com/catalyst/moodle-mod_cms/pull/92/files#diff-53c6c2616dd05e8d131ed7cc22c92e2d18fccbc96a79f7923e2c6604c78672a7R162
https://github.com/catalyst/moodle-mod_cms/pull/92/files#diff-53c6c2616dd05e8d131ed7cc22c92e2d18fccbc96a79f7923e2c6604c78672a7R176